src/Cache/InMemoryDriver.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace Tnapf\Config\Cache;
+
+use DateInterval;
+use Psr\SimpleCache\CacheInterface;
+use Psr\SimpleCache\InvalidArgumentException;
+use Tnapf\Config\Exceptions\InvalidCacheKeyException;
+
+class InMemoryDriver implements CacheInterface
+{
+    private array $storage = [];
+
+    /**
+     * @throws InvalidArgumentException
+     */
+    public function get(string $key, mixed $default = null): mixed
+    {
+        return $this->has($key) ? $this->storage[$key] : $default;
+    }
+
+    /**
+     * @throws InvalidCacheKeyException
+     */
+    public function set(string $key, mixed $value, DateInterval|int|null $ttl = null): bool
+    {
+        $this->validateKey($key);
+        $this->storage[$key] = $value;
+
+        return true;
+    }
+
+    /**
+     * @throws InvalidCacheKeyException
+     */
+    public function delete(string $key): bool
+    {
+        $this->validateKey($key);
+        unset($this->storage[$key]);
+
+        return true;
+    }
+
+    public function clear(): bool
+    {
+        $this->storage = [];
+
+        return true;
+    }
+
+    /**
+     * @param string[] $keys
+     *
+     * @throws InvalidArgumentException
+     *
+     * @return iterable<string, mixed>
+     */
+    public function getMultiple(iterable $keys, mixed $default = null): iterable
+    {
+        $items = [];
+        foreach ($keys as $key) {
+            $items[$key] = $this->get($key, $default);
+        }
+
+        return $items;
+    }
+
+    /**
+     * @throws InvalidArgumentException
+     */
+    public function setMultiple(iterable $values, DateInterval|int|null $ttl = null): bool
+    {
+        foreach ($values as $key => $value) {
```

Comment:
Maybe a bit jank, but would this work?
```php
$this->storage = [...$this->storage, ...$values];
```
Guessing no, cuz of iterable instead of array

Reply:
This works, but then there's no `validateKey` call.

Reply:
It's ever so slightly faster to do it with my suggestion and a separate loop for validating keys. But its ~0.01 to 0.03 seconds faster for every million uses so not worth it